tools/update_runnablejobs.py

```diff
@@ -191,59 +260,88 @@ def add_jobs_to_jobpriority(new_data=None, priority=1, timeout=0, set_expired=Fa
+        _buildsystem = job["build_system_type"]
+        key = _unique_key(job)
+        if key in map:
+            # We already know about this job, we might need to update the build system
+            row_build_system_type = map[key]['build_system_type']
+
+            if row_build_system_type == '*' or _buildsystem == '*':
+                # We don't need to update anything
+                pass
+            else:
+                # We're seeing the job again but for another build system (e.g. buildbot vs
+                # taskcluster). We need to change it to '*'
+                if row_build_system_type != _buildsystem:
                     _buildsystem = "*"
-                    found_id = row[0]
 
-        # We have new jobs from runnablejobs to add to our master list
-        if not found:
-            _expired = None
-            if set_expired:
-                # set _expired = today + 14 days
-                # TODO: write test for it
-                _expired = "%s" % (datetime.datetime.now() + datetime.timedelta(days=14))
+                # Compare original build system to the new one
+                if _buildsystem != job['build_system_type']:
+                    # XXX: Is this if check worth keeping? or should we join we the one above?
+                    #      Where we set _buildsystem to '*'
+                    # Update table with new buildsystem
+                    conn = engine.connect()
+                    statement = update(JobPriorities).where(
+                        JobPriorities.id == map[key]['pk_key']).values(buildsystem=_buildsystem)
+                    conn.execute(statement)
+                    LOG.info('Updated {} from {} to {}'.format(
+                        job['platform'], job['build_system_type'], _buildsystem
+                    ))
+                    updated_jobs += 1
 
+        else:
+            # We have a new job from runnablejobs to add to our master list
             try:
-                jobpriority = JobPriorities(str(testtype),
-                                            str(job["platform_option"]),
-                                            str(job["build_platform"]),
-                                            priority,
-                                            timeout,
-                                            _expired,
-                                            _buildsystem)
-
+                jobpriority = JobPriorities(
+                    str(job["testtype"]),
+                    str(job["platform_option"]),
+                    str(job["platform"]),
+                    priority,
+                    timeout,
+                    expiration_date,
+                    _buildsystem
+                )
                 session.add(jobpriority)
                 session.commit()
-                added_jobs.append(job)
             except Exception as error:
                 session.rollback()
-                logging.warning(error)
+                LOG.warning(error)
```

Comment:
I've seen this information on the logs.
We actually get to see the sql warning.